Plugins/Public/mobiledocking_plugin/Main.cpp

```diff
@@ -460,97 +792,100 @@ int __cdecl Dock_Call(unsigned int const &iShip, unsigned int const &iBaseID, in
-				{
-					uint iDockedClientID = HkGetClientIdFromCharname(i->second);
-
-					// Update the coordinates the given ship should launch to.
-					UpdateCarrierLocationInformation(iDockedClientID, cship->get_position(), cship->get_orientation());
-
-					// Carrier is no more. Set the flag.
-					mobiledockClients[iDockedClientID].carrierDied = true;
-
-					// Due to the carrier not existing anymore, we have to pull the system information from the carriers historical location.
-					mobiledockClients[iDockedClientID].carrierSystem = cship->iSystem;
-
-					JettisonShip(client, iDockedClientID);
-				}
-
-				// Clear the carrier from the list
-				mobiledockClients[client].mapDockedShips.clear();
-				mobiledockClients[client].iDockingModulesAvailable = mobiledockClients[client].iDockingModulesInstalled;
-			}
+			PrintUserCmdText(carrierID, L"%s has left your garrison, reclaiming hangar space.", charName);
 		}
+
+		RemoveShipFromLists(charName, false);
 	}
 }
 
-
-bool UserCmd_Process(uint client, const wstring &wscCmd)
+bool UserCmd_Process(uint client, const wstring& wscCmd)
 {
 	returncode = DEFAULT_RETURNCODE;
 
 	if (wscCmd.find(L"/listdocked") == 0)
 	{
-		if (mobiledockClients[client].mapDockedShips.empty())
+		if (!idToCarrierInfoMap.count(client) || idToCarrierInfoMap[client]->dockedShipList.empty())
 		{
-			PrintUserCmdText(client, L"No ships currently docked");
+			PrintUserCmdText(client, L"No ships currently docked, free capacity: %u", mobiledockClients[client].iDockingModulesAvailable);
+			return true;
 		}
-		else
+
+		uint counter = 0;
+		PrintUserCmdText(client, L"Docked ships:");
+		for (wstring& dockedShip : idToCarrierInfoMap[client]->dockedShipList)
 		{
-			PrintUserCmdText(client, L"Docked ships:");
-			for (map<wstring, wstring>::iterator i = mobiledockClients[client].mapDockedShips.begin();
-				i != mobiledockClients[client].mapDockedShips.end(); ++i)
+			counter++;
+			uint dockedClientID = HkGetClientIdFromCharname(dockedShip.c_str());
+			if (dockedClientID != -1)
+			{
+				const auto& shipInfo = Archetype::GetShip(Players[dockedClientID].iShipArchetype);
+				wstring& shipName = HkGetWStringFromIDS(shipInfo->iIdsName);
+				uint ship;
+				pub::Player::GetShip(dockedClientID, ship);
+				if (ship)
+				{
+					PrintUserCmdText(client, L"%u. %ls - %ls - LAUNCHED", counter, dockedShip.c_str(), shipName.c_str());
+				}
+				else
+				{
+					PrintUserCmdText(client, L"%u. %ls - %ls - ON STANDBY", counter, dockedShip.c_str(), shipName.c_str());
+				}
+			}
+			else
 			{
-				PrintUserCmdText(client, i->first);
+				PrintUserCmdText(client, L"%u. %ls - OFFLINE", counter, dockedShip.c_str());
 			}
+			counter++;
+		}
+		if (mobiledockClients[client].iDockingModulesAvailable)
+		{
+			PrintUserCmdText(client, L"Remaining free capacity: %u", mobiledockClients[client].iDockingModulesAvailable);
 		}
 		return true;
 	}
 	else if (wscCmd.find(L"/conn") == 0 || wscCmd.find(L"/return") == 0)

```

Comment:
kinda feel like this should be done with inter-plugin communication rather than trying to process the user's commands first but I recognise it's pre-existing code